core/base-service/node-fetch.js

```diff
@@ -0,0 +1,101 @@
+import { URL, URLSearchParams } from 'url'
+import fetch from 'node-fetch'
+import { Inaccessible, InvalidResponse } from './errors.js'
+
+const userAgent = 'Shields.io/2003a'
+
+function object2URLSearchParams(obj) {
+  const qs = {}
+  for (const [key, value] of Object.entries(obj)) {
+    if (value === undefined) {
+      continue
+    } else if (value === null) {
+      qs[key] = ''
+    } else if (['string', 'number', 'boolean'].includes(typeof value)) {
+      qs[key] = value
+    }
+  }
+  return new URLSearchParams(qs)
+}
+
+function request2NodeFetch({ url, options }) {
+  const requestOptions = Object.assign({}, options)
+  const nodeFetchOptions = {}
+  const nodeFetchUrl = new URL(url)
+  const interchangableOptions = ['headers', 'method', 'body']
+
+  if ('body' in requestOptions && 'form' in requestOptions) {
+    throw new Error("Options 'form' and 'body' can not both be used")
+  }
+
+  interchangableOptions.forEach(function (opt) {
+    if (opt in requestOptions) {
+      nodeFetchOptions[opt] = requestOptions[opt]
+      delete requestOptions[opt]
+    }
+  })
+  nodeFetchOptions.headers = nodeFetchOptions.headers || {}
+
+  if ('qs' in requestOptions) {
+    if (typeof requestOptions.qs === 'string') {
+      nodeFetchUrl.search = requestOptions.qs
+      delete requestOptions.qs
+    } else if (typeof requestOptions.qs === 'object') {
+      nodeFetchUrl.search = object2URLSearchParams(requestOptions.qs)
+      delete requestOptions.qs
+    } else if (requestOptions.qs == null) {
+      delete requestOptions.qs
+    } else {
+      throw new Error("Property 'qs' must be string, object or null")
+    }
+  }
+
+  if ('gzip' in requestOptions) {
+    nodeFetchOptions.compress = requestOptions.gzip
+    delete requestOptions.gzip
+  }
+
+  if ('auth' in requestOptions) {
+    const user = requestOptions.auth.user || ''
+    const pass = requestOptions.auth.pass || ''
+    const b64authStr = Buffer.from(`${user}:${pass}`).toString('base64')
+    nodeFetchOptions.headers.Authorization = `Basic ${b64authStr}`
+    delete requestOptions.auth
+  }
+
+  if ('form' in requestOptions) {
+    nodeFetchOptions.body = object2URLSearchParams(requestOptions.form)
```

Comment:
There are 2 ways you can do form encoding: URL Encoded and Multipart I've implemented URL Encoded here. This:
- is simpler to implement
- is easier to mock with nock
- works with all the services we've got that are sending forms

If we wanted to do multipart encoding instead, that implementation would look like

```js
import FormData from 'form-data'

const form = new FormData()
for (const [key, value] of Object.entries(requestOptions.form)) {
    form.append(key, value)
}
nodeFetchOptions.body = form
const formHeaders = form.getHeaders()
nodeFetchOptions.headers['content-type'] = formHeaders['content-type']
delete requestOptions.form
```

One thing I haven't really got my head round is how higher level clients decide or allow the user to specify when to use each encoding type. Given URL Encoding works for everything we're currently doing, I reckon we can leave this how it is and worry about multipart in future if we move forward with node-fetch and find a use-case that needs it.